### PR TITLE
ref(notifications): Document Notification Platform

### PR DESCRIPTION
**Pull request by mgaeta**
The Ecosystem Team is starting to add more documentation to the Notification Platform. In this PR we create helpers for making URLs and footers.

### CODE REVIEW

File: src/sentry/integrations/slack/message_builder/footer.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    if provider == ExternalProviders.SLACK:
+        return f"<{link.url}|{link.label}>"
+    raise RuntimeError("Invalid Provider")
+
+
+def process_footer_part(
+    notification: BaseNotification,
+    recipient: Team | User,
+    provider: ExternalProviders,
+    part: FooterLink | str,
```
Comment (ceorourke) [inline src/sentry/integrations/slack/message_builder/footer.py:34]: If `part` is a link can you call this variable `link`?

File: src/sentry/integrations/slack/message_builder/footer.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        return build_link(footer_link, provider)
+
+    return part
+
+
+def build_footer(
+    notification: BaseNotification,
+    recipient: Team | User,
+    provider: ExternalProviders,
+    parts: Sequence[str],
```
Comment (ceorourke) [inline src/sentry/integrations/slack/message_builder/footer.py:52]: What are `parts`?

File: src/sentry/integrations/slack/message_builder/footer.py (hunk 1 above)
Comment (mgaeta) [reply src/sentry/integrations/slack/message_builder/footer.py:34]: `part` can be a string or a link.

File: src/sentry/integrations/slack/message_builder/footer.py (hunk 2 above)
Comment (mgaeta) [reply src/sentry/integrations/slack/message_builder/footer.py:52]: `parts` are strings or links.

File: src/sentry/integrations/slack/message_builder/notifications.py [hunk 3]
```diff
@@ -1,98 +0,0 @@
-from __future__ import annotations
```
Comment (ceorourke) [inline src/sentry/integrations/slack/message_builder/notifications.py:1]: Why delete this file and make it 3? This PR is difficult to review because you have to mentally ignore the "noise" and try to find the actual changes which take up the minority of the diff from what I can tell. Can you remove the unrelated things and just have a PR with the changes?

File: tests/sentry/integrations/slack/notifications/test_deploy.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 from sentry.testutils.helpers.slack import get_attachment, send_notification
 
 
-class SlackUnassignedNotificationTest(SlackActivityNotificationTest):
+class SlackDeployNotificationTest(SlackActivityNotificationTest):
```
Comment (ceorourke) [inline tests/sentry/integrations/slack/notifications/test_deploy.py:12]: Good catch but these 7 test file changes don't seem related to the rest of the PR.

File: src/sentry/integrations/slack/message_builder/notifications.py (hunk 3 above)
Comment (scefali) [reply src/sentry/integrations/slack/message_builder/notifications.py:1]: Yes, that would make reviewing much easier

File: src/sentry/integrations/slack/message_builder/footer.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        environment = latest_event.get_environment()
+    except Environment.DoesNotExist:
+        return None
+
+    environment_name: str | None = getattr(environment, "name", None)
+    return environment_name
+
+
+def get_role_string(member: OrganizationMember) -> str:
+    if member.role == "billing":
```
Comment (scefali) [inline src/sentry/integrations/slack/message_builder/footer.py:77]: @mgaeta we shouldn't have references to getsentry roles in Sentry, we should keep it separate. That's why this method was on a class method that could be overwitten

**Review by scefali — CHANGES_REQUESTED** [review]

File: src/sentry/notifications/utils/urls.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+def get_user_settings_url(type_key: NotificationSettingTypes | None = None) -> str:
+    if not type_key:
+        return "/settings/account/notifications/"
+
+    fine_tuning_key = get_notification_setting_type_name(type_key)
+    return f"/settings/account/notifications/{fine_tuning_key}/"
+
+
+def get_settings_url(
```
Comment (scefali) [inline src/sentry/notifications/utils/urls.py:102]: @mgaeta Why are we reversing the change of making this a class method on the notification class? What's the point of doing it this way where you need `notification` instance as the first argument when you could always just reference `self`? I know we are currently not overriding this method on a child class, but maybe in the future we might.

File: src/sentry/integrations/slack/message_builder/footer.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            footer_link = FooterLink(
+                label="Notification Settings",
+                url=get_settings_url(notification, recipient, provider),
+            )
+        return build_link(footer_link, provider)
+
+    return part
+
+
+def build_footer(
```
Comment (scefali) [inline src/sentry/integrations/slack/message_builder/footer.py:48]: @mgaeta where is `build_footer` used?

File: src/sentry/integrations/slack/message_builder/footer.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    recipient: Team | User,
+    provider: ExternalProviders,
+    parts: Sequence[str],
+) -> str:
+    return " | ".join(
+        [process_footer_part(notification, recipient, provider, part) for part in parts]
+    )
+
+
+def get_environment_name(group: Group | None) -> str | None:
```
Comment (scefali) [inline src/sentry/integrations/slack/message_builder/footer.py:59]: @mgaeta and how is `get_environment_name` used?
